src/geometry/MergedMeshGroup.ts

```diff
@@ -158,11 +159,29 @@ export class MergedMesh {
   fileId: number;
   treeIndexMap: { [s: number]: number };
   createdByInstancedMesh: boolean;
-  constructor(capacity: number, fileId: number, createdByInstancedMesh: boolean = false) {
+  geometry?: THREE.BufferGeometry;
+  diffuseTexture?: TextureInfo;
+  specularTexture?: TextureInfo;
+  ambientTexture?: TextureInfo;
+  normalTexture?: TextureInfo;
+  bumpTexture?: TextureInfo;
+  constructor(capacity: number, fileId: number,
+    createdByInstancedMesh: boolean = false,
+    diffuseTexture?: TextureInfo,
```

Comment:
will the amount of textures increase? should we make this an object so its not as large as a constructor :P

Reply:
No, should be these for the time being. But I totally agree. If there are more incoming at some point, we can create a separate object.